### ISSUE TITLE
Can't get get_user_entered_format() to work

### ISSUE THREAD

**Issue by ArielS1** [robin900/gspread-formatting#11]
I'm trying to find out a specific cell's color like this:
format = get_user_entered_format(worksheet, "R2C1")

and i get the following exception:
Traceback (most recent call last): File ".\script.py", line 25, in <module> parser() File ".\script.py", line 23, in script format = get_user_entered_format(worksheet, "R2C1") File "C:\Python37\lib\site-packages\gspread_formatting.py", line 89, in get_user_entered_format props = resp['sheets'][0]['data'][0]['rowData'][0]['values'][0].get('userEnteredFormat') KeyError: 'rowData'

the worksheet is a valid worksheet generated by gc.open_by_url('https://..').sheet1
and the cell exists and has a background color

**Comment by robin900** [comment]
Try `get_effective_format` first — you may get the format you expect that way.

**Comment by ArielS1** [comment]
I run get_effective_format(worksheet, "R2C1")
and get:
`Traceback (most recent call last):
  File ".\script.py", line 28, in <module>
    parser()
  File ".\s.py", line 25, in parser
    foramt = get_effective_format(worksheet, "R2C1")
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 65, in get_effective_format
    props = resp['sheets'][0]['data'][0]['rowData'][0]['values'][0].get('effectiveFormat')
KeyError: 'rowData'`

**Comment by robin900** [comment]
Well, that is weird. Are you sure there is a format in place for that cell?

**Comment by ArielS1** [comment]
Yes, maybe I got a wrong range?
The cell is in row 2, column A
I got the label from what that cell prints so I manually formatted a string that is "R2C1"
Am I doing something wrong?

**Comment by robin900** [comment]
Try just `A2`

**Comment by ArielS1** [comment]
When i use : foramt = get_effective_format(worksheet, "A2")
i get:
`Traceback (most recent call last):
  File ".\script.py", line 28, in <module>
    parser()
  File ".\script.py", line 25, in parser
    foramt = get_effective_format(worksheet, "A2")
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 66, in get_effective_format
    return CellFormat.from_props(props) if props else None
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 215, in from_props
    return _props_to_component(_underlower(cls.__name__), props)
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 180, in _props_to_component
    v = _props_to_component(k, v)
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 180, in _props_to_component
    v = _props_to_component(k, v)
  File "C:\Python37\lib\site-packages\gspread_formatting.py", line 176, in _props_to_component
    raise ValueError("No format component named '%s'" % class_alias)
ValueError: No format component named 'top'`

**Comment by robin900** [comment]
Try the following debugging code to output the parsed JSON response for the above call to `get_effective_format(worksheet, "A2")`. Then post the output to this issue.

```
resp = worksheet.spreadsheet.fetch_sheet_metadata({
        'includeGridData': True,
        'ranges': ['%s!%s' % (worksheet.title, 'A2')],
        'fields': 'sheets.data.rowData.values.effectiveFormat'
    })
props = resp['sheets'][0]['data'][0]['rowData'][0]['values'][0].get('effectiveFormat')
print(props)
```

**Comment by ArielS1** [comment]
this is the output:
{  
   'backgroundColor':{  
      'red':0.7764706,
      'green':0.9372549,
      'blue':0.80784315
   },
   'borders':{  
      'top':{  
         'style':'SOLID',
         'width':1,
         'color':{  

         }
      },
      'bottom':{  
         'style':'SOLID',
         'width':1,
         'color':{  

         }
      },
      'left':{  
         'style':'SOLID',
         'width':1,
         'color':{  

         }
      },
      'right':{  
         'style':'SOLID',
         'width':1,
         'color':{  

         }
      }
   },
   'padding':{  
      'top':2,
      'right':3,
      'bottom':2,
      'left':3
   },
   'horizontalAlignment':'CENTER',
   'verticalAlignment':'BOTTOM',
   'wrapStrategy':'OVERFLOW_CELL',
   'textFormat':{  
      'foregroundColor':{  

      },
      'fontFamily':'arial,sans,sans-serif',
      'fontSize':12,
      'bold':True,
      'italic':False,
      'strikethrough':False,
      'underline':False
   },
   'hyperlinkDisplayType':'PLAIN_TEXT'
}

**Comment by robin900** [comment]
@ArielS1 I think I've deduced your problem -- you are using version <= 0.0.4 of the package. Version 0.0.5 includes a fix for certain cases of CellFormat parsing; version 0.0.6 includes a fix for corner cases when a certain object in the CellFormat is an empty object. Upgrade your package to 0.0.6 and you should be fine. Please let me know how version 0.0.6 works for you.

**Comment by ArielS1** [comment]
I installed this package 2 days ago, so it must be the latest version.
I also have this path installed:
C:\Python37\Lib\site-packages\gspread_formatting-0.0.6.dist-info
so i guess it's 0.0.6

**Comment by robin900** [comment]
@ArielS1 The reason I suspected your release to be <0.0.6 is that your stacktrace shows an old version of the module code. But I've trace the problem to a mistake in my setup.py that caused the version 0.0.6 release to contain old code!

I've released version 0.0.7 to PyPI just now, and I've verified that it has the new code. With that release, your troubles should vanish.

**Comment by robin900** [comment]
Fixed by release 0.0.7.
